pkg/artifactory/provider/sdkv2.go

```diff
@@ -76,23 +73,33 @@ func SdkV2() *schema.Provider {
 func providerConfigure(ctx context.Context, d *schema.ResourceData, terraformVersion string) (interface{}, diag.Diagnostics) {
 	tflog.Debug(ctx, "providerConfigure")
 
-	URL := d.Get("url")
+	// Check environment variables, first available OS variable will be assigned to the var
+	url := CheckEnvVars([]string{"JFROG_URL", "ARTIFACTORY_URL"}, "http://localhost:8082")
+	accessToken := CheckEnvVars([]string{"JFROG_ACCESS_TOKEN", "ARTIFACTORY_ACCESS_TOKEN"}, "")
 
-	restyBase, err := client.Build(URL.(string), productId)
+	if d.Get("url") != "" {
+		url = d.Get("url").(string)
+	}
+	if d.Get("access_token") != "" {
+		accessToken = d.Get("access_token").(string)
+	}
+	apiKey := d.Get("api_key").(string)
+
+	restyBase, err := client.Build(url, productId)
 	if err != nil {
 		return nil, diag.FromErr(err)
 	}
 
-	apiKey := d.Get("api_key").(string)
-	accessToken := d.Get("access_token").(string)
-
 	restyBase, err = client.AddAuth(restyBase, apiKey, accessToken)
 	if err != nil {
 		return nil, diag.FromErr(err)
 	}
-
+	// Due to migration from SDK v2 to plugin framework, we have to remove defaults from the provider configuration.
+	// https://discuss.hashicorp.com/t/muxing-upgraded-tfsdk-and-framework-provider-with-default-provider-configuration/43945
+	// License check will happen if `check_license` is true or if it's not set.
 	checkLicense := d.Get("check_license").(bool)
-	if checkLicense {
+	_, checkLicenseBoolSet := d.GetOk("check_license")
```

Comment:
Line #100 is redundant. Just have it as

```go
checkLicense, ok := d.GetOk("check_license")
if (ok && checkLicense) || !ok {
...
}
```

It's a bit cryptic though 🤔

Reply:
Exactly, I've done it to be very clear :)